### PR TITLE
embeddings refactor with NPZ bert loads

### PR DESCRIPTION
**Pull request by dpressel**
This change provides a converter from huggingface transformers to NPZ format which can be used by Transformer LMs and fine-tuning directly using the MEAD Transformer layers.  Once the change is accepted, these converted checkpoints will be uploaded to make it easy to use.

Also this change eliminates the need for a BERT-specific addon -- now BERT is just a special case of `embed_tlm_*` where the layer norms happen after the layers and the embeddings include an LN reduction.  

As part of the change, it was easiest to refactor the embeddings to follow the same approach as the
pytorch.  As part of that, some issues were identified and fixed with the naming.

There is now a mixin to supply the additonal required interfaces for TF.  The models now subclass the original layers and the mixin to get state recording.

As part of the change, calls to shallow copy the embeddings sub-graphs were removed -- it doesnt appear that these are required at the moment, although we will run multi-GPU testing to confirm.

One note: The `embed_tlm` doesnt (cannot?) support custom token type embeddings at the momemt.  Its possible to use these but I think it probably requires extending the model itself, not the embedding.  In the future we probably should support multiple features per embedding

### CODE REVIEW

**Comment by dpressel** [comment]
Closes #546 and #528

File: baseline/tf/embeddings.py [hunk 1]
```diff
@@ -120,116 +82,86 @@ def get_config(self):
...
         return tf.compat.v1.placeholder(tf.int32, [None, None, None], name=name)
 
 
 @register_embeddings(name='char-transformer')
-class CharTransformerModel(TensorFlowEmbeddingsModel):
-    def __init__(self, name=None, **kwargs):
-        super().__init__(name, **kwargs)
-        self.embedding_layer = CharTransformerEmbeddings(name=self._name, **kwargs)
+class CharTransformerModel(CharTransformerEmbeddings, TensorFlowEmbeddingsMixin):
+    pass
```
Comment (blester125) [inline baseline/tf/embeddings.py:102]: Why does this embedding not need a create placeholder?

File: baseline/tf/embeddings.py [hunk 2]
```diff
@@ -26,49 +24,18 @@ def detached_ref(self):
...
         self.x = x
-        return self.embedding_layer(x)
-
-    def call(self, x):
-        return self.encode(x)
 
-    def get_feed_dict(self):
-        """Return a feed dict that is needed to initialize this embeddings."""
-        return self.embedding_layer.get_feed_dict()
+        return super().encode(x)
```
Comment (blester125) [inline baseline/tf/embeddings.py:38]: I thought part of the plan of simplifying the embedding was to remove the `encode` and replace it with `call`?

File: layers/eight_mile/tf/layers.py [hunk 3]
```diff
@@ -1792,7 +1795,7 @@ def items(self):
     def __getitem__(self, item: str):
         return self.embeddings[item]
 
-    def call(self, inputs):
+    def call(self, inputs: Dict[str, tf.Tensor]) -> tf.Tensor:
```
Comment (blester125) [inline layers/eight_mile/tf/layers.py:1798]: I thought we were using that `TensorDef` for type hints about tensors now?

**Pull request merged**